Mathlib/Topology/CWComplex/Classical/Subcomplex.lean

```diff
@@ -157,12 +159,344 @@ instance RelCWComplex.Subcomplex.finite_subcomplex_of_finite [T2Space X] [RelCWC
+      exact union_eq_right.symm)
+
+instance RelCWComplex.Subcomplex.instDistribLattice [RelCWComplex C D] :
+    DistribLattice (Subcomplex C) :=
+  {(instLattice : Lattice (Subcomplex C)) with
+    le_sup_inf E F G := by
+      simp only [← SetLike.coe_subset_coe, coe_max, coe_min]
+      rw [← union_inter_distrib_left]
+    }
+
+@[simps -isSimp]
+instance RelCWComplex.Subcomplex.instTop [T2Space X] [RelCWComplex C D] : Top (Subcomplex C) where
+  top := {
+    carrier := C
+    I n := univ
+    closed' := isClosed
+    union' := by simp [← union_iUnion_openCell_eq_complex]
+  }
+
+@[simps -isSimp]
+instance RelCWComplex.Subcomplex.instBot [RelCWComplex C D] : Bot (Subcomplex C) where
+  bot := {
+    carrier := D
+    I n := ∅
+    closed' := isClosedBase C
+    union' := by simp
+  }
+
+instance RelCWComplex.Subcomplex.finite_bot [T2Space X] [RelCWComplex C D] :
+    Finite ((⊥ : Subcomplex C) : Set X) where
+  eventually_isEmpty_cell := by simp [cell_def, bot_I]
+  finite_cell n := by simp [cell_def, bot_I n, Finite.of_subsingleton]
+
+@[simps! -isSimp]
+instance RelCWComplex.Subcomplex.instSupSet [T2Space X] [RelCWComplex C D] :
+    SupSet (Subcomplex C) where
+  sSup S := mk' C (D ∪ ⋃ (E ∈ S), E) (fun n ↦ ⋃ (E ∈ S), E.I n)
+    (by
+      intro n ⟨i, hi⟩
+      apply subset_union_of_subset_right
+      simp_rw [mem_iUnion] at hi
+      obtain ⟨E, hE, hEj⟩ := hi
+      apply subset_iUnion_of_subset E
+      apply subset_iUnion_of_subset hE
+      rw [← E.union_closedCell (C := C)]
+      apply subset_union_of_subset_right
+      exact subset_iUnion_of_subset n (subset_iUnion
+        (fun (j : ↑(E.I n)) ↦ closedCell (C := C) n ↑j) ⟨i, hEj⟩))
+    (by
+      simp_rw [← Subcomplex.union]
+      rw [← iUnion_subtype (p := fun E ↦ E ∈ S)
+        (s := fun E ↦ D ∪ ⋃ n, ⋃ (j : E.1.I n), openCell n j.1)]
+      by_cases h : Nonempty S
+      · rw [← union_iUnion, ← union_assoc, union_self, iUnion_comm]
+        congrm D ∪ ?_
+        apply iUnion_congr fun n ↦ ?_
+        simp_rw [iUnion_subtype, mem_iUnion, iUnion_exists]
+        rw [iUnion_comm]
+        apply iUnion_congr fun E ↦ ?_
+        rw [iUnion_comm]
+      · simp_all)
+
+lemma RelCWComplex.Subcomplex.iSup_carrier [T2Space X] [RelCWComplex C D]
+    {J : Type*} (f : J → Subcomplex C) :
+    ((⨆ (j : J), f j) : Subcomplex C) = D ∪ ⋃ j, f j := by
+  simp [iSup, coe_sSup]
+
+@[simp]
+lemma RelCWComplex.Subcomplex.iSup_coe_eq_of_nonempty [T2Space X] [RelCWComplex C D]
+    {J : Type*} [Nonempty J] (f : J → Subcomplex C) :
+    (((⨆ (j : J), f j) : Subcomplex C) : Set X) = ⋃ j, f j := by
+  simp only [iSup_carrier, union_eq_right]
+  apply subset_iUnion_of_subset (Classical.choice (α := J) inferInstance)
+  exact base_subset_complex
+
+@[simp]
+lemma RelCWComplex.Subcomplex.iSup_coe_eq_of_empty [T2Space X] [RelCWComplex C D]
+    {J : Type*} [IsEmpty J] (f : J → Subcomplex C) :
+    ⨆ (j : J), f j = ⊥ := by
+  ext x
+  simp [← SetLike.mem_coe, iSup_carrier, coe_bot]
```

Comment:
I don't think this lemma is needed: once you get the proper instances it is a general fact about complete lattices.